lua/dylib.lua

```diff
@@ -30,6 +71,81 @@ int add_fib_entry(const char *prefix, const char *gt_ip, const char *gw_ip,
 	enum gk_fib_action action, struct gk_config *gk_conf);
 int del_fib_entry(const char *ip_prefix, struct gk_config *gk_conf);
 
+struct ipaddr *get_fib_dump_entry_ip(struct gk_fib_dump_entry *entry);
+struct ipaddr *get_fib_dump_entry_grantor_ip(struct gk_fib_dump_entry *entry);
+struct ipaddr *get_fib_ether_dump_entry_nexthop(
+	struct gk_fib_ether_dump_entry *entry);
+struct ether_addr *get_fib_ether_dump_entry_mac(
+	struct gk_fib_ether_dump_entry *entry);
+int convert_ip_to_str(struct ipaddr *ip_addr, char *res, int n);
+void ether_format_addr(char *buf, uint16_t size,
+	const struct ether_addr *eth_addr);
+
 ]]
 
 c = ffi.C
+
+function print_fib_dump_entry(fib_dump_entry, arg)
+	local i = 0
+	local ret
+	local max_ip_addr_len = 128
+	local entry = ffi.cast("struct gk_fib_dump_entry *", fib_dump_entry)
```

Comment:
Once `luaL_pushcdata()` is used in C, this cast won't be necessary.

Reply:
This comment hasn't been addressed after the big bug was sorted out.

Reply:
To make it work correctly, I have to use ``lua_pushlightuserdata()``. In this case, if we removed the cast, there is an error: ``./lua/dylib.lua:96: attempt to index global 'fib_dump_entry' (a userdata value)``